tests/prefetch_related/models.py

```diff
@@ -323,3 +323,23 @@ class Flea(models.Model):
     )
     pets_visited = models.ManyToManyField(Pet, related_name="fleas_hosted")
     people_visited = models.ManyToManyField(Person, related_name="fleas_hosted")
+
+
+class Related(models.Model):
+    pass
+
+
+class GrandParent(models.Model):
+    name = models.CharField(max_length=50)
+    gp_fk = models.ForeignKey(
+        Related, null=True, on_delete=models.CASCADE, related_name="gp_fk_rel"
+    )
+    gp_m2m = models.ManyToManyField(Related, related_name="gp_m2m_rel")
+
+
+class Parent(GrandParent):
+    pass
+
+
+class Child(Parent):
+    pass
```

Comment:
Any reason why `AuthorWithAge` can't be used here by creating a third base `AuthorWithAgeChild(AuthorWithAge)`?

I get that these models make it crystal clear what exactly we're testing but can't create 5 (one extra for the m2m) tables each time we identity a bug as it compounds towards slower test execution.

Maybe a compromise could be to reuse `Author` but not use `AuthorWithName` but `AuthorChild(Author)` and `AuthorNestedChild(AuthorChild)` so we only add two tables while allowing `Book` and `Author` to be used to test foreign key and many to many fields.

Reply:
Makes sense. I honestly did not even think about test execution time, but it absolutely makes sense in a project as large as Django. I've changed the test to use `Author` and `AuthorWithAge` and only added one model, `AuthorWithAgeChild`. I was able to reuse the existing foreign key and m2m relations in Author.